lib/Config.js

```diff
@@ -269,6 +269,28 @@ class Config {
             }
         }
 
+        this.metadataDaemon = { host: 'localhost', port: 9990 };
+        if (config.metadataDaemon) {
+            if (config.metadataDaemon.host !== undefined &&
+                config.metadataDaemon.port !== undefined) {
+                assert.strictEqual(
+                    typeof config.metadataDaemon.host, 'string',
+                    'bad config: metadata daemon host must be a string');
+                this.metadataDaemon.host = config.metadataDaemon.host;
+
+                assert(Number.isInteger(config.metadataDaemon.port)
+                       && config.metadataDaemon.port > 0,
+                       'bad config: metadata daemon port must be a ' +
+                       'positive integer');
+                this.metadataDaemon.port = config.metadataDaemon.port;
+            } else {
+                assert(config.metadataDaemon.host === undefined &&
+                       config.metadataDaemon.port === undefined,
+                       'bad config: you must specify both host and ' +
+                       'port of metadata daemon or neither of them');
```

Comment:
I don't think you need this assertion and the condition `if` above because the assertions of `host` and `port` in that `if` does the job of the whole block. Just need to move all the assignment to after the assertions.